File: prueba/Servidor/Core/integradorrecaudo/src/main/java/integradorrecaudo/usuario/dto/UsuarioDTO.java
package integradorrecaudo.usuario.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Data;
import lombok.experimental.FieldDefaults;

import java.util.Set;

@Builder
@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UsuarioDTO {

    Integer id;
    @JsonProperty("nombre_usuario")
    String nombreUsuario;

    String nombres;

    String apellidos;
    String correo;
    String contrasena;

    String documento;
    boolean activo;
    Set<String> roles;
}

File: prueba/Cliente/Web/IntegradorRecaudo/src/app/data/schema/usuarioDto.ts
import { GenericDto } from './GenericDto';
import { RolDto } from './RolDto';

export class UsuarioDto extends GenericDto {
  id?:number;
  nombre_usuario?: string;
  contrasena?: string;
  estado?: string;
  nombres?: string;
  apellidos?: string;
  correo?: string;
  documento?: string;
  id_tipo_documento?: number;
  roles?: String[];

  constructor(init?: Partial<UsuarioDto>) {
    super();
    if (init) {
      Object.assign(this, init);
    }
  }
}
